### ISSUE TITLE
Segfault building headercvt

### ISSUE THREAD

**Issue by pradyparanjpe** [fixstars/clpy#267]
# Problem:
I'm trying to install clpy from git source. (pip repo appears to be a placeholder.)
I've compiled llvm-clang 9.0.1 (latest) locally in `${HOME}/.local`.
(dnf package manager has llvm/clang too, but I'm not using it as recommended)
**Setup throws segfault.**

`$ cd ${HOME}/programs/clpy`
`$ python3 ./setup.py` -> output follows:

'''
building ultima started
building headercvt
clang++ -std=c++11 -D_GLIBCXX_USE_CXX11_ABI=1 -Wall -Wextra -O3 -pedantic-errors ultima.cpp -lclangTooling -lclangFrontendTool -lclangFrontend -lclangDriver -lclangSerialization -lclangCodeGen -lclangParse -lclangSema -lclangStaticAnalyzerFrontend -lclangStaticAnalyzerCheckers -lclangStaticAnalyzerCore -lclangAnalysis -lclangARCMigrate -lclangRewrite -lclangEdit -lclangAST -lclangLex -lclangBasic -lclang `llvm-config --libs --system-libs` -fno-rtti -o ultima
clang++ -std=c++11 -D_GLIBCXX_USE_CXX11_ABI=1 -Wall -Wextra -O3 -pedantic-errors headercvt.cpp -lclangTooling -lclangFrontendTool -lclangFrontend -lclangDriver -lclangSerialization -lclangCodeGen -lclangParse -lclangSema -lclangStaticAnalyzerFrontend -lclangStaticAnalyzerCheckers -lclangStaticAnalyzerCore -lclangAnalysis -lclangEdit -lclangAST -lclangLex -lclangBasic -lclang `llvm-config --libs --system-libs` -fno-rtti -o headercvt
make: *** [Makefile:2: ultima] Segmentation fault (core dumped)
make: *** [Makefile:4: headercvt] Segmentation fault (core dumped)
Traceback (most recent call last):
  File "./setup.py", line 8, in <module>
    import clpy_setup_build
  File "${HOME}/programs/clpy/clpy_setup_build.py", line 38, in <module>
    raise RuntimeError('Building headercvt has been failed.')
RuntimeError: Building headercvt has been failed.
'''

### Platform:
`$ uname -a` -> output:
Linux *$HOSTNAME* 5.4.17-200.fc31.x86_64 #1 SMP Sat Feb 1 19:00:13 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

`$ lspci` -> output:
...
00:02.0 VGA compatible controller: Intel Corporation HD Graphics 5500 (rev 09)
...
08:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Opal XT [Radeon R7 M265/M365X/M465]
...

`$ lscpu` -> output:
...
Model name:                      Intel(R) Core(TM) i7-5500U CPU @ 2.40GHz
...

### exports:
`export C_INCLUDE_PATH="${C_INCLUDE_PATH}:/usr/include/CL";`
`CPLUS_INCLUDE_PATH="${CPLUS_INCLUDE_PATH}:/usr/include/CL";`
`LIBRARY_PATH="${LIBRARY_PATH}:/usr/lib64";`

`# export PATH="${PATH}"; # LLVM Binaries are in already exported ${HOME}/.local/bin`
`export CPLUS_INCLUDE_PATH="${CPLUS_INCLUDE_PATH}:${HOME}/.local/include";`
`export LIBRARY_PATH="${LIBRARY_PATH}:${HOME}/.local/lib:${HOME}/.local/lib64";`
`export LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:${HOME}/.local/lib:${HOME}/.local/lib64";`

Since it was always looking in /usr/lib directory, I had to symlink
`$ ln -s ${HOME}/.local/lib/libclangTooling.a  /usr/lib/libclangTooling.a`

**Comment by vorj** [comment]
@pradyparanjpe Thank you for your reporting!
We have confirmed that ClPy works with LLVM/Clang 9.0.1, and the builds are succeeded in our environment, the _secondary machine_ . (And now I'm checking on the _primary machine_ too).

Could you try the LLVM/Clang on DNF? If the SEGVs remain, please comment it again.

EDIT(15:58 JST): We confirmed that the builds of headercvt and ultima are succeeded using hand-compiled LLVM/Clang 9.0.1 on _primary machine_ .

**Comment by pradyparanjpe** [comment]
I had already installed clang and llvm using dnf for some other compilations in the past. llvm-clang that I now compiled locally, was probably never used. (Hence, the necessity of the alias mentioned in the previous comment)
I uninstalled the dnf-installed llvm-clang and compiled clpy.

Note: Latest python3.8.x has numpy 1.18.x. Clpy issues compilation of numpy<=1.13.3, that fails with python3.8 (probably because of a variable rename, I don't know exactly), however, I successfully installed clpy in a python3.6 (as mentioned in README.md) on a virtualenv with numpy1.13.3.
